jOOQ/src/main/java/org/jooq/tools/Convert.java

```diff
@@ -380,7 +380,7 @@ else if (from[0] != null && from[0].getClass() == toComponentType) {
      * @throws DataTypeException - When the conversion is not possible
      */
     public static final <T> T convert(Object from, Class<? extends T> toClass) throws DataTypeException {
-        return ConvertAll.from(ConvertAll.from(from, toClass), toClass);
+        return ConvertAll.from(from, toClass);
```

Comment:
This change has been applied on top of your own change, so it is difficult to assess from the change itself whether it is correct.

I do remember there's a case where this was needed, but I can't track it down to an individual regression test right now